gateway-ha/src/test/java/io/trino/gateway/ha/persistence/TestDatabaseMigrationsOracle.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.persistence;
+
+import org.jdbi.v3.core.Handle;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.OracleContainer;
+
+public class TestDatabaseMigrationsOracle
+        extends BaseTestDatabaseMigrations
+{
+    @Override
+    protected final JdbcDatabaseContainer<?> startContainer()
+    {
+        JdbcDatabaseContainer<?> container = new OracleContainer("gvenzl/oracle-xe:18.4.0-slim");
+        container.start();
+        return container;
+    }
+
+    @Override
+    protected final String getDriver()
+    {
+        return "oracle.jdbc.driver.OracleDriver";
+    }
+
+    @Override
+    public void testMigrationWithExistingGatewaySchema()
+    {
+        // We do not need to run this test with Oracle since Oracle
+        // backend DB support did not exist before so there
+        // can be no existing deployments of gateway using
+        // Oracle as the backend database.
+    }
+
+    @Override
+    protected void dropAllTables()
+    {
+        String gatewayBackendTable = "DROP TABLE gateway_backend";
+        String queryHistoryTable = "DROP TABLE query_history";
+        String propertiesTable = "DROP TABLE resource_groups_global_properties";
+        String resourceGroupsTable = "DROP TABLE resource_groups";
+        String selectorsTable = "DROP TABLE selectors";
+        String exactMatchTable = "DROP TABLE exact_match_source_selectors";
+        String flywayHistoryTable = "DROP TABLE \"flyway_schema_history\"";
+        Handle jdbiHandle = jdbi.open();
+        String sql = String.format("SELECT 1 FROM all_tables WHERE owner = '%s'", getTestSchema());
+        //String sql = String.format("SELECT 1 FROM USER_OBJECTS WHERE OBJECT_TYPE = 'table'");
+        verifyResultSetCount(sql, 7);
+        jdbiHandle.execute(gatewayBackendTable);
+        jdbiHandle.execute(queryHistoryTable);
+        jdbiHandle.execute(propertiesTable);
+        jdbiHandle.execute(selectorsTable);
+        jdbiHandle.execute(resourceGroupsTable);
+        jdbiHandle.execute(exactMatchTable);
+        jdbiHandle.execute(flywayHistoryTable);
+        verifyResultSetCount(sql, 0);
+        jdbiHandle.close();
```

Comment:
This should make it a little easier to update this function when tables are added in the future
```suggestion
        List<String> tables = ImmutableList.of("gateway_backend", "query_history","resource_groups_global_properties", "resource_groups", "selectors", "exact_match_source_selectors", "\"flyway_schema_history\"");
        String sql = String.format("SELECT 1 FROM all_tables WHERE owner = '%s'", getTestSchema());
        verifyResultSetCount(sql, tables.size());
        tables.forEach(t -> jdbiHandle.execute(String.format("DROP TABLE %s", t)));
        verifyResultSetCount(sql, 0);
        jdbiHandle.close();
```